docs/Integration_Guide.md

```diff
@@ -0,0 +1,115 @@
+    - [For immutable contracts (inline protection):](#for-immutable-contracts-inline-protection)
+    - [For mutable contracts (proxy protection):](#for-mutable-contracts-proxy-protection)
+    - [In the deployment Scripts](#in-the-deployment-scripts)
+
+
+## Protection  - the big picture
+- Spherex protection works by identifiying suspicious and malicious behavior patterns of a transaction as it is being executed by the protocol (between any number of contracts)
+- The granularity of the protection is **functions** - meaning the smallest unit of code we track behaviour of is a function.
+  - The data collected and used for each function in each protocol may change, depending on various factors (gas cost in the chain, past history, client requests, etc.)
+- `pure` and `view` functions are not protected, as they do not change the state of the contract , and therefore do not require protection.
+
+## Some terms and defenitions
+
+#### Contracts in the system
+
+- **Protected Contract**: The smart contract that is being protected by Spherex.
+- **Spherex Engine**: This is a smart contract written by spherex that acts as *the brain* of the protection system.
+  - The protected contracts in the protocol communicate with the Spherex Engine to perform various operations (send information and get approval for the current state of the transaction).
+
+#### Entities and addresses
+
+- **Operator**: The address (whether EOA or a contract) that actaully configures and updates the protection of the contract. This can be either the protocol developer/manager, the SphereX tech team, or any other wallet that has the permission to do so.
+- **Admin**: The address (whether EOA or a contract) that controls the operator. This should be the protocol owner , and it's meant as a protection against a rogue operator.
+- **Allowed Senders** - One of our protection mechanisms is to only allow pre approved addresses to send data to the engine. So this is just a list of all the contracts that are allowed to send data to the engine.
+  - This becomes important if you deploy contracts dynamically (for example, using the factory pattern).
+
+This is a good place to emphesize and reiterate: the **Operator** and the **Admin** can be a wallet address (EOA) but cal also be a smart contract (like a DAO contract or a multisig)
+
+This also means that if indeed we use a smart contract as the operator or the admin, we will need to set the various roles accordingly, and also implement in that *managine contract* the various managment functions needed:
+- An Admin should be able to set the operator
+- An Operator has many many funcionalities (like setting the engine, adding and removing allowed senders and patterns, and even configuring the active rules) - Please talk to the SphereX team to help you understand the various roles and permissions.
+
+If the protocol has some kind of a registry or management contract, it might be a good candidate for the operator/admin roles.
```

Comment:
So if someone takes control of management, it also controls the protection?

Reply:
yes. He who controls the spice, controls the universe